esx_service/vmdk_ops.py

```diff
@@ -690,26 +690,36 @@ def detachVMDK(vmdk_path, vm_uuid):
 
 # Check existence (and creates if needed) the path for docker volume VMDKs
 def get_vol_path(datastore, tenant_name=None):
-    # If the command is NOT running under a tenant, the folder for Docker
+    # If tenant_name is set to None, the folder for Docker
     # volumes is created on <datastore>/DOCK_VOLS_DIR
-    # If the command is running under a tenant, the folder for Dock volume
+    # If tenant_name is set, the folder for Dock volume
     # is created on <datastore>/DOCK_VOLS_DIR/tenant_uuid
     # a symlink <datastore>/DOCK_VOLS_DIR/tenant_name will be created to point to
     # path <datastore>/DOCK_VOLS_DIR/tenant_uuid
+    # If the dock volume folder already exists,
+    # the path returned contains tenant name not UUID.
+    # This is to make logs more readable. OS will resolve this path
+    # as a symlink with tenant_name will already be present.
 
-    path = dock_vol_path = os.path.join("/vmfs/volumes", datastore, DOCK_VOLS_DIR)
+    readable_path = path = dock_vol_path = os.path.join("/vmfs/volumes", datastore, DOCK_VOLS_DIR)
 
     if tenant_name:
         error_info, tenant = auth_api.get_tenant_from_db(tenant_name)
         if error_info:
             logging.error("get_vol_path: failed to find tenant info for tenant %s", tenant_name)
             path = dock_vol_path
         path = os.path.join(dock_vol_path, tenant.id)
+        readable_path = os.path.join(dock_vol_path, tenant_name)
 
-    if os.path.isdir(path):
-        # If the path exists then return it as is.
-        logging.debug("Found %s, returning", path)
-        return path, None
+    if os.path.isdir(readable_path) or os.path.isdir(path):
```

Comment:
I just feel the new logic here is different from old ones...
Before this fix, if path is not a dir, it will go ahead to the following steps to create it or run necessary things.
But the new logic is, as long as the readable_path is there (even path is not a dir), it will never go to the rest of the code.
Is it possible that, the original path is getting deleted but the symlink path is still existing. Then the original path will never be created again?
I am not sure if this is a possible situation. Just raise the concern.
So I think it's better to keep this check as it was "if os.path.isdir(path):", but add your following checks.